services/steam/steam.service.js

```diff
@@ -0,0 +1,395 @@
+const { age: ageColor } = require('../../lib/color-formatters')
+const prettyBytes = require('pretty-bytes')
+
+const steamCollectionSchema = Joi.object({
+  response: Joi.object()
+    .keys({
+      result: Joi.number()
+        .integer()
+        .min(1)
+        .max(1)
+        .required(),
+      resultcount: Joi.number()
+        .integer()
+        .min(0)
+        .max(1)
+        .required(),
+      collectiondetails: Joi.array()
+        .items(
+          Joi.object({
+            publishedfileid: Joi.string(),
+            result: Joi.number().integer(),
+            children: Joi.array(),
+          })
+        )
+        .required(),
+    })
+    .required(),
+}).required()
+
+const steamFileSchema = Joi.object({
+  response: Joi.object()
+    .keys({
+      result: Joi.number()
+        .integer()
+        .min(1)
+        .required(),
+      resultcount: Joi.number()
+        .integer()
+        .min(0)
+        .required(),
+      publishedfiledetails: Joi.array().items(
+        Joi.object({
+          publishedfileid: Joi.string().required(),
+          result: Joi.number()
+            .integer()
+            .required(),
+          file_size: Joi.number().integer(),
+          time_created: Joi.number().integer(),
+          subscriptions: Joi.number().integer(),
+          favorited: Joi.number().integer(),
+          lifetime_subscriptions: Joi.number().integer(),
+          lifetime_favorited: Joi.number().integer(),
+          views: Joi.number().integer(),
+        })
+      ),
+    })
+    .required(),
+}).required()
+
+class SteamCollectionFiles extends BaseJsonService {
+  async fetch({ collectionId }) {
+    const url =
+      'https://api.steampowered.com/ISteamRemoteStorage/GetCollectionDetails/v1/?format=json'
+    return this._requestJson({
+      url,
+      schema: steamCollectionSchema,
+      errorMessages: {
+        400: 'bad request',
+      },
+      options: {
+        method: 'POST',
+        form: {
+          collectioncount: '1',
+          'publishedfileids[0]': collectionId,
+        },
+      },
+    })
+  }
+
+  static render({ size }) {
+    return { message: metric(size), color: 'green' }
+  }
+
+  async handle({ collectionId }) {
+    const json = await this.fetch({ collectionId })
+    if (json.response.collectiondetails[0].result === 1) {
+      return this.constructor.render({
+        size: json.response.collectiondetails[0].children.length,
+      })
+    } else {
+      return { message: 'collection not found', color: 'red' }
+    }
+  }
+
+  static get category() {
+    return 'other'
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'files' }
+  }
+
+  static get url() {
+    return {
+      base: 'steam/collection-files',
+      format: '(.+)',
+      capture: ['collectionId'],
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Steam Collection Files',
+        exampleUrl: '180077636',
+        urlPattern: ':collection_id',
```

Comment:
How does a user get or work out their collection ID or file ID? Is this really clea/obvious to users in this ecosystem or do we need to document how to obtain these IDs?

Reply:
The id is in the URL of the workshop item.
Example https://steamcommunity.com/sharedfiles/filedetails/?id=1525218777 => `?id=1525218777`

How to tell the other users of this I don't know how to do that.